libraries/AC_WPNav/AC_Circle.cpp

```diff
@@ -98,7 +98,7 @@ void AC_Circle::init()
 /// set_circle_rate - set circle rate in degrees per second
 void AC_Circle::set_rate(float deg_per_sec)
 {
-    if (!is_equal(deg_per_sec,_rate)) {
+    if (!is_equal(deg_per_sec,(float)_rate)) {
```

Comment:
@dgrat Sorry I dont understand the comment? However using forwarding functions you can guide the arguments towards the single implementation function you want. No need for explicit casting in application code, which is one major goal is it not? The other goal being to avoid duplicate code. "code bloat" This achieves both objectives.The forwarding function should be defined inline in the header since it doesnt do much except fixing up the arguments. The implementation function should not be inlined of course if it involves a substantial amount of code. This style of code can easily be optimised for size and speed to match non-template code